build failure: `python3Packages.flaks-mail`

### Describe the bug
I can't build `pgadmin4` when declared like `services.pgadmin.enable = true`, but I can build it with `nix build nixpkgs#pgadmin`. It seems `python312.flask` package doesn't pass all the tests:
```
python3.12-flask-mail> tests/test_message.py ...............................FF.....             [100%]
python3.12-flask-mail> =================================== FAILURES ===================================
python3.12-flask-mail> __________________________ test_unicode_sender_tuple ___________________________
python3.12-flask-mail> app = <Flask 'conftest'>, mail = <flask_mail.Mail object at 0x7ffff56e3c20>
python3.12-flask-mail>     def test_unicode_sender_tuple(app: Flask, mail: Mail) -> None:
python3.12-flask-mail>         msg = Message(
python3.12-flask-mail>             subject="subject",
python3.12-flask-mail>             sender=("ÄÜÖ → ✓", "from@example.com>"),
python3.12-flask-mail>             recipients=["to@example.com"],
python3.12-flask-mail>         )
python3.12-flask-mail> >       assert (
python3.12-flask-mail>             "From: =?utf-8?b?w4TDnMOWIOKGkiDinJM=?= <from@example.com>" in msg.as_string()
python3.12-flask-mail>         )
python3.12-flask-mail> E       assert 'From: =?utf-8?b?w4TDnMOWIOKGkiDinJM=?= <from@example.com>' in 'Content-Type: text/plain; charset="utf-8"\r\nMIME-Version: 1.0\r\nContent-Transfer-Encoding: 7bit\r\nSubject: subject...e.com\r\nDate: Thu, 17 Oct 2024 19:14:41 +0000\r\nMessage-ID: <172919248166.157.9918194149393024129@localhost>\r\n\r\n'
python3.12-flask-mail> E        +  where 'Content-Type: text/plain; charset="utf-8"\r\nMIME-Version: 1.0\r\nContent-Transfer-Encoding: 7bit\r\nSubject: subject...e.com\r\nDate: Thu, 17 Oct 2024 19:14:41 +0000\r\nMessage-ID: <172919248166.157.9918194149393024129@localhost>\r\n\r\n' = as_string()
python3.12-flask-mail> E        +    where as_string = <flask_mail.Message object at 0x7ffff56e02f0>.as_string
python3.12-flask-mail> tests/test_message.py:423: AssertionError
python3.12-flask-mail> _____________________________ test_unicode_sender ______________________________
python3.12-flask-mail> app = <Flask 'conftest'>, mail = <flask_mail.Mail object at 0x7ffff5734bf0>
python3.12-flask-mail>     def test_unicode_sender(app: Flask, mail: Mail) -> None:
python3.12-flask-mail>         msg = Message(
python3.12-flask-mail>             subject="subject",
python3.12-flask-mail>             sender="ÄÜÖ → ✓ <from@example.com>>",
python3.12-flask-mail>             recipients=["to@example.com"],
python3.12-flask-mail>         )
python3.12-flask-mail> >       assert (
python3.12-flask-mail>             "From: =?utf-8?b?w4TDnMOWIOKGkiDinJM=?= <from@example.com>" in msg.as_string()
python3.12-flask-mail>         )
python3.12-flask-mail> E       assert 'From: =?utf-8?b?w4TDnMOWIOKGkiDinJM=?= <from@example.com>' in 'Content-Type: text/plain; charset="utf-8"\r\nMIME-Version: 1.0\r\nContent-Transfer-Encoding: 7bit\r\nSubject: subject....com\r\nDate: Thu, 17 Oct 2024 19:14:41 +0000\r\nMessage-ID: <172919248175.157.17296093936275749371@localhost>\r\n\r\n'
python3.12-flask-mail> E        +  where 'Content-Type: text/plain; charset="utf-8"\r\nMIME-Version: 1.0\r\nContent-Transfer-Encoding: 7bit\r\nSubject: subject....com\r\nDate: Thu, 17 Oct 2024 19:14:41 +0000\r\nMessage-ID: <172919248175.157.17296093936275749371@localhost>\r\n\r\n' = as_string()
python3.12-flask-mail> E        +    where as_string = <flask_mail.Message object at 0x7ffff5734650>.as_string
python3.12-flask-mail> tests/test_message.py:435: AssertionError
python3.12-flask-mail> =========================== short test summary info ============================
python3.12-flask-mail> FAILED tests/test_message.py::test_unicode_sender_tuple - assert 'From: =?utf-8?b?w4TDnMOWIOKGkiDinJM=?= <from@example.com>' in 'Cont...
python3.12-flask-mail> FAILED tests/test_message.py::test_unicode_sender - assert 'From: =?utf-8?b?w4TDnMOWIOKGkiDinJM=?= <from@example.com>' in 'Cont...
python3.12-flask-mail> ========================= 2 failed, 49 passed in 0.80s =========================
```

i've also tried installing without it beeing a service (`environment.systemPackages = [pkgs.pgadmin4];`), with the same luck. Also i've tried disabling test checks and the email server, same logs:
```nix
services.pgadmin = {
  enable = true;
  package = pkgs.pgadmin4.overrideAttrs (finalAttrs: previousAttrs: {
    doCheck = false;
    doInstallCheck = false;
  });
  initialPasswordFile = ./initial-pgadmin-pass;
  initialEmail = "Benvenute!";
  emailServer.enable = false;
};
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. rebuilding my nixos system

### Expected behavior
to be able to build the package

### Notify maintainers

@gador 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.54, NixOS, 24.11 (Vicuna), 24.11.20241006.c31898a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - nixpkgs: `/nix/store/rs4fjbnw4qx7ns2hzzrz2iz52va7vs5z-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/flask-mail/default.nix
+++ b/pkgs/development/python-modules/flask-mail/default.nix
@@ -31,6 +31,13 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [ pytestCheckHook ];
 
+  disabledTests = [
+    # Broken by fix for CVE-2023-27043.
+    # Reported upstream in https://github.com/pallets-eco/flask-mail/issues/233
+    "test_unicode_sender_tuple"
+    "test_unicode_sender"
+  ];
+
   meta = {
     description = "Flask extension providing simple email sending capabilities";
     homepage = "https://github.com/pallets-eco/flask-mail";

